graylog2-server/src/main/java/org/graylog2/indexer/indexset/CustomFieldMappings.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.indexer.indexset;
+
+import com.google.common.collect.ImmutableMap;
+import org.graylog2.indexer.fieldtypes.FieldTypeMapper;
+import org.graylog2.indexer.fieldtypes.FieldTypes;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CustomFieldMappings extends HashSet<CustomFieldMapping> {
```

Comment:
Is there a reason why we extend the full HashSet class? Do we really want to provide all set methods that are implemented in the hashset?

Reply:
We do want to be able to treat this as a `Set<CustomFieldMapping>` elsewhere, for iteration, etc. - so it makes sense to use a `HashSet` for the implementation. We do not need to implement any of the methods. There is IMO no smalles interface that we could implement without extending from something.

Reply:
ok, makes sense, thanks!